### PR TITLE
feat: Upgrade backstage to version 1.6.0 

### PR DESCRIPTION
**Pull request by SamoKopecky**
Part of #86

Upgrade backstage from `1.2.2` to `1.6.0`. After some consideration I decided it would be easier to just upgrade in one big PR instead of 4 small PRs.

Upgrade helper: https://backstage.github.io/upgrade-helper/?from=1.2.2&to=1.6.0

I tested the built image on my dev cluster, everything seemed to work fine. For reference the image from this PR is deployed [here](https://service-catalog-service-catalog.apps.skopecky.dev.datahub.redhat.com/) (access from RedHat network only).

## Major changes
- Version `1.0.0` of the search plugins brings a big rework to the search functionality. Nothing changes much on the surface but instead of the [Lunr search engine](https://backstage.io/docs/features/search/search-engines#lunr) (configured [here](https://github.com/operate-first/service-catalog/blob/main/packages/backend/src/plugins/search.ts#L16-L18)) we can use [Postgres search engine](https://backstage.io/docs/features/search/search-engines#postgres). This will be added in a separate PR (#149), until that PR is merged the search functionality won't work properly.
- Kubernetes plugin now queries 2 more resources (`limitranges`, `daemonsets`), see [this PR](https://github.com/operate-first/apps/pull/2545/files).
- During the build process this error will pop up:
```shell
$ backstage-cli package build
app:  
app:  Loaded config from app-config.yaml, app-config.local.yaml
app:  
app:  /bin/sh: line 1: git: command not found
app:  
app:  WARNING: Failed to read git commit, ExitCodeError: Command 'git rev-parse HEAD' exited with code 127
app:  
app:  /bin/sh: line 1: git: command not found
app:  
app:  WARNING: Failed to describe git version, ExitCodeError: Command 'git describe --always' exited with code 127
```
since we don't have git in our base image. I haven't discovered any effect this might have on the final image.
- Bunch of updated package versions

/cc @tumido

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -16,41 +16,42 @@
...
+    "@backstage/plugin-permission-node": "^0.6.5",
+    "@backstage/plugin-proxy-backend": "^0.2.30",
+    "@backstage/plugin-scaffolder-backend": "^1.6.0",
+    "@backstage/plugin-search-backend": "^1.0.2",
+    "@backstage/plugin-search-backend-module-pg": "^0.4.0",
+    "@backstage/plugin-search-backend-node": "^1.0.2",
+    "@backstage/plugin-techdocs-backend": "^1.3.0",
     "@internal/plugin-cluster-status-backend": "*",
     "app": "link:../app",
+    "better-sqlite3": "^7.5.0",
```
Comment (tumido) [inline packages/backend/package.json:42]: Why to move this to deps from dev deps? In prod we use external DB anyways...

**Comment by SamoKopecky** [comment]
The techdocs feature isn't necessary since we are not using [edit_url](https://www.mkdocs.org/user-guide/configuration/#edit_uri) in any of our `mkdocs.yaml` files and this feature would require this value to be set.

**Comment by SamoKopecky** [comment]
There still seem to be more problems with the search, figuring it out.

/hold

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**